sequencer.c

```diff
@@ -4386,6 +4386,34 @@ int append_todo_help(unsigned edit_todo, unsigned keep_empty)
 	return ret;
 }
 
+int edit_todo_list(unsigned flags)
+{
+	struct strbuf buf = STRBUF_INIT;
+	const char *todo_file = rebase_path_todo();
+	FILE *todo;
+
+	if (strbuf_read_file(&buf, todo_file, 0) < 0)
+		return error(_("could not read '%s'."), todo_file);
+
+	strbuf_stripspace(&buf, 1);
+	todo = fopen_or_warn(todo_file, "w");
+	if (!todo)
+		return 1;
+
+	strbuf_write(&buf, todo);
+	fclose(todo);
+	strbuf_release(&buf);
+
+	transform_todos(flags | TODO_LIST_SHORTEN_IDS);
+	append_todo_help(1, 0);
+
+	launch_editor(todo_file, NULL, NULL);
```

Comment:
What happens if launching the editor fails? Does launch_editor() die or return an error? Which will be the exit code of git-rebase--interactive.sh?

Reply:
`launch_editor()` returns an error, but it won’t be handled, so the return code will be 0.

Reply:
The original shell script was:

```
		git_sequence_editor "$todo" ||
			die "$(gettext "Could not execute editor")" 
```

so the exit code would not have been 0 with the shell code.

Reply:
Also, `transform_todos()` returns an error, but its errors were ignored by git-rebase--interactive.sh. Should I handle them?

Reply:
Same question for `append_todo_help()`.

Reply:
I think for now it is ok if the errors that were ignored are still ignored.

Reply:
Or maybe it would be an obvious bonus of the rewrite ;-)